### PR TITLE
Update some show commands on StarOS table

### PR DESCRIPTION
**Pull request by wyb**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Problem Summary(Required) ：

Update the following commands:
1. Show tablet tablet_id
2. Show tablet from table, add ShardId
3. Show partitions for table, add UseStarOS
4. Show index proc, add UseStarOS
5. Statistic proc, do not count the replica number of StarOS tablet

### CODE REVIEW

**Comment by wyb** [comment]
run starrocks_fe_unittest

**Comment by wyb** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 51 / 94 (54.26%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/StatisticProcDir.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/common/proc/IndicesProcDir.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/common/proc/TabletsProcDir.java|48|83|57.83%|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|2|2|100.00%|

**Comment by wyb** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 60 / 94 (63.83%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/StatisticProcDir.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/common/proc/IndicesProcDir.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/common/proc/TabletsProcDir.java|57|83|68.67%|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|2|2|100.00%|

**Comment by wyb** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 84 / 94 (89.36%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/StatisticProcDir.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/common/proc/IndicesProcDir.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/common/proc/TabletsProcDir.java|81|83|97.59%|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|2|2|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/common/proc/StatisticProcDir.java [hunk 1]
```diff
@@ -129,14 +129,20 @@ public ProcResult fetchResult() throws AnalysisException {
...
                         short replicationNum = olapTable.getPartitionInfo().getReplicationNum(partition.getId());
                         ++dbPartitionNum;
                         for (MaterializedIndex materializedIndex : partition
                                 .getMaterializedIndices(IndexExtState.VISIBLE)) {
                             ++dbIndexNum;
                             for (Tablet tablet : materializedIndex.getTablets()) {
-                                LocalTablet localTablet = (LocalTablet) tablet;
                                 ++dbTabletNum;
+
+                                if (useStarOS) {
```
Comment (wuleistarrocks) [inline fe/fe-core/src/main/java/com/starrocks/common/proc/StatisticProcDir.java:141]: dbReplicaNum ignores StarOS Tablet, is this right?

**Comment by wyb** [comment]
run starrocks_be_unittest

File: fe/fe-core/src/main/java/com/starrocks/common/proc/StatisticProcDir.java (hunk 1 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/common/proc/StatisticProcDir.java:141]: Yes, i think this refers to the storage replicas of tablet, but StarOS tablet only has computing replicas.
We may add more columns to distinguish between storage replicas and computing replicas later.

**Review by gengjun-git — DISMISSED** [review]

**Review by wuleistarrocks — DISMISSED** [review]

**Comment by sduzh** [comment]
IMO, before StarOS is stable and publicly released we should try to keep the user interfaces unchanged. @imay How do you think about this?

**Comment by wyb** [comment]
> IMO, before StarOS is stable and publicly released we should try to keep the user interfaces unchanged. @imay How do you think about this?
Updated.
The StarOS columns will be showed only if `Config.use_staros` is true.

**Review by sduzh — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 102 / 123 (82.93%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/StatisticProcDir.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/common/proc/IndicesProcDir.java|0|9|00.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|10|12|83.33%|
|:large_blue_circle:|com/starrocks/common/proc/TabletsProcDir.java|91|96|94.79%|

**Review by gengjun-git — DISMISSED** [review]

**Review by sduzh — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 102 / 123 (82.93%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/StatisticProcDir.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/common/proc/IndicesProcDir.java|0|9|00.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|10|12|83.33%|
|:large_blue_circle:|com/starrocks/common/proc/TabletsProcDir.java|91|96|94.79%|

**Review by gengjun-git — APPROVED** [review]

**Review by wuleistarrocks — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Pull request merged**